src/common/SubProcess.cc

```diff
@@ -8,11 +8,19 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <iostream>
+#include <iomanip>
+
 
 #include "common/errno.h"
 #include "include/ceph_assert.h"
 #include "include/compat.h"
 
+#ifdef _WIN32
+#define INVALID_PID_VALUE 0
+#else
+#define INVALID_PID_VALUE -1
+#endif
```

Comment:
what's the point of defining this? also you never compare `pid` with `INVALID_PID_VALUE`. i think the better way to implement this is not to hide the difference between POSIX and Win32, and use `int` for representing PID on both platforms. better off using `DWORD` on Win32 and use `int` on POSIX.

actually, i'd suggest create a `SubProcess_win32.h` and `SubProcess_win32.cc` for Win32 port.

Reply:
We initialize the pid with ``INVALID_PID_VALUE`` in a few places.

> actually, i'd suggest create a SubProcess_win32.h and SubProcess_win32.cc for Win32 port.

I'd rather not waste any more time on this, seems like a nit that wasn't brought up before. How about a TODO inline comment? We're moving way too slow, mostly stumbling upon nits.

Reply:
>  seems like a nit that wasn't brought up before

because `INVALID_PID_VALUE` was added in the latest revision. 

> We're moving way too slow, mostly stumbling upon nits.

okay. i will stop reviewing this changeset then. i cannot approve the changes i don't understand.

Reply:
It's assigned in the following places, one of them being the constructor which is used on both platforms:

https://github.com/ceph/ceph/blob/9d2d06fd07d8cc78eeb5d1b0eb6cc9423662b534/src/common/SubProcess.cc#L33
https://github.com/ceph/ceph/blob/9d2d06fd07d8cc78eeb5d1b0eb6cc9423662b534/src/common/SubProcess.cc#L438

Here's another place where we can specifically use that value: https://github.com/ceph/ceph/blob/9d2d06fd07d8cc78eeb5d1b0eb6cc9423662b534/src/common/SubProcess.cc#L517

Let me know if this requires further explanation.

Reply:
I've just moved the Windows bits to a separate file called SubProcess_win32.cc, dropping ``INVALID_PID_VALUE`` as well.

About the header, I'd prefer using the same file for now, if that's acceptable. The alternative would be to add an abstract class, have the SubProcessPosix/SubProcessWin32 concrete classes and rename them to SubProcess based on the OS, hiding this detail from the callers. Still, that might delay this PR even more.

PS @tchaikov I'm sorry if any of the previous comments seemed offensive, that was not my intention. I really do appreciate the reviews, it's just that we're getting pressed for time and I'm doing my best to keep things moving.